cpp/src/arrow/dataset/scanner.h

```diff
@@ -188,6 +185,7 @@ class ARROW_DS_EXPORT Scanner {
 
  protected:
   std::shared_ptr<Dataset> dataset_;
+  std::shared_ptr<Fragment> fragment_;
```

Comment:
Can you add a comment explaining that this is a temporary solution until ARROW-8065 is resolved so it doesn't confuse reader.

Reply:
done